corehq/apps/app_manager/views/releases.py

```diff
@@ -122,16 +123,32 @@ def _get_batch(start_build=None, skip=None):
         )
         if only_show_released:
             app_es = app_es.is_released()
-        if build_comment:
-            app_es = app_es.build_comment(build_comment)
+        if query:
+            app_es = app_es.build_comment(query)
+
         results = app_es.exclude_source().run()
+        total_apps = results.total
         app_ids = results.doc_ids
         apps = get_docs(Application.get_db(), app_ids)
+
+        # Check if user is searching for a specific version number
+        if query:
+            try:
+                version = int(query)
+            except ValueError:
+                pass
+            else:
+                build_doc = get_build_doc_by_version(domain, app_id, version)
```

Comment:
One last query,
other apps are searched by ES and this one by couch view and hence different fields in this doc. You already pointed in the PR description. in this case,
1. The web page would break if some field is missing in this result from couch?
2. Or we would need to maintain fields in both couch view and ES result for this to keep working?

can we wait for @esoergel feedback on using ES to find the build by version?

Reply:
I'm not aware of us having problems with docs or fields missing from couch - couch has been reliable, just slow. What has caused problems in the past with this code is that docs have been missing from ES. So it's more likely that the search might miss a document, especially a recent one, but because search is a less-used use case and because couch-based search is so painfully slow, we can live with that.

Reply:
ethan's suggestion might make this irrelevant but still sharing.

sorry, what i meant was the result docs from ES would have different attributes of app doc than what we would get from Couch.
Agree on missing from ES. The one and only instance i have heard of that was when one makes a build and it does not show when the page is reloaded. i think that was changed back to Couch as seen in `_get_batch` in case of no build_comment/only_show_released.

Reply:
I'm going to look at implementing Ethan's feedback, but I'm curious, why would ES and couch have different fields? If ES is behind?

Reply:
I think Manish is referring to the transformation that sometimes happens before sending docs to ES.  This is usually the addition of a few calculated properties for certain types of queries.  For instance, we insert `owner_type` into case documents.

For apps, this is that transformation:
https://github.com/dimagi/commcare-hq/blob/master/corehq/pillows/application.py#L16-L19
That is, it applies lazy migrations, but these migrations aren't applied to docs pulled from couch unless they're wrapped.

Reply:
Thanks @esoergel